.travis.yml

```diff
@@ -58,5 +49,4 @@ deploy:
       tags: true
       all_branches: true
 
-after_deploy:
-- "echo deployed"
+after_deploy: "echo deployed"
```

Comment:
I wonder if it is worth.
Yes it is shorter but less readable.  In the expanded form, I can immediately see which platforms are built.  In the shorter view, I have to know more about travis-ci file.

Reply:
Since all of the builds except macos are executed in the exact same environment from tavis' point of view, I think it makes sense to have them defined as default, and use the `jobs include` to underline differences.

However, the most useful feature is that adding a different build is just one line:

```
  - IMAGE=fedora26+llvm
```

See [.travis.yml#L20-L31](https://github.com/1138-4EB/ghdl/blob/08494d68ecd2c0c2df60c901ec545f6c2a296f72/.travis.yml#L20-L31) in #489.

Since travis does not allow to dinamically change the jobs matrix (that was the original motivation to spin all the runs in a single travis job), being able to add/remove builds (un)commenting a single line makes it easier to have 4-5 platforms for everyday runs, but do complete tests less frequently (say, before a release).

Reply:
OK, but worth adding a comment to explain how that works.

Reply:
How about this?

> Defining the same environment variable in the top-level `env` field is interpreted by travis as a 'matrix expansion'. I.e., a separate job is run for each of the given values. Then, `jobs:include` is used to include additional jobs, besides the expanded matrix. See:
- tgingold/ghdl#488
- https://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix
- https://docs.travis-ci.com/user/build-stages/#Build-Stages-and-Build-Matrix-Expansion

Reply:
I think you simply need to describe what you are doing; no need to refer to the documentation.
Something like:
The linux builds share the same environment and are factored out; however the mac build is described as an additional job.

Reply:
Done. I reordered the content to leave all the common constraints above the matrix (i.e. I moved the deploy section).
Note that linux builds do not exactly 'share the same environment', an independent machine/box is used for each job/build, but all of them are based on the same constraints/environment.